frontend/src/lib/helpers/market-stats.tsx

```diff
@@ -0,0 +1,137 @@
+import { cn } from '@/lib/utils';
+import {
+  formatCurrency,
+  formatPrice,
+  formatSupply,
+} from '@/lib/utils/format-number';
+import { getTimeAgo } from '@/lib/utils/format-time';
+import { calculateTokenScore } from '@/lib/utils/market-stats';
+import type { TokenWithEngagement } from '@/types/token-monitor/token';
+import type { TweetWithAnalysis } from '@/types/tweets-analysis';
+import type { ReactNode } from 'react';
+
+interface ScoreBarProps {
+  score: number;
+}
+
+function ScoreBar({ score }: ScoreBarProps) {
+  return (
+    <div className="flex items-center gap-2">
+      <div className="w-8 h-1 bg-emerald-400/20 rounded-full overflow-hidden">
+        <div
+          className={cn(
+            'h-full rounded-full',
+            score >= 80
+              ? 'bg-emerald-400'
+              : score >= 60
+                ? 'bg-emerald-400/80'
+                : score >= 40
+                  ? 'bg-emerald-400/60'
+                  : score >= 20
+                    ? 'bg-emerald-400/40'
+                    : 'bg-emerald-400/20',
+          )}
+          style={{ width: `${score}%` }}
+        />
+      </div>
+      <span>{score.toFixed(1)}</span>
+    </div>
+  );
+}
+
+export const renderValue = (
```

Comment:
imo only this should be in the helper, the name should be different too IMO, and the component extracted.